lib/install/management/appliance.go

```diff
@@ -219,6 +222,69 @@ func (d *Dispatcher) deleteVM(vm *vm.VirtualMachine, force bool) error {
 		d.op.Warnf("Failed to remove datastore files for VM path %q: %s", folder, err)
 	}
 
+	// at this point the VCH should be gone and we need to cleanup the inventory
+	err = d.deleteVCHInventoryFolders()
+	if err != nil {
+		// if this fails manual cleanup is necessary
+		return err
+	}
+
+	return nil
+}
+
+func (d *Dispatcher) deleteVCHInventoryFolders() error {
```

Comment:
This could be moved to `lib/install/management/delete.go` instead of being here. Thoughts?

Reply:
Needs to be. Separation of concerns.